stackslib/src/net/p2p.rs

```diff
@@ -5301,20 +5305,35 @@ impl PeerNetwork {
             let mut new_stackerdb_configs = HashMap::new();
             let stacker_db_configs = mem::replace(&mut self.stacker_db_configs, HashMap::new());
             for (stackerdb_contract_id, stackerdb_config) in stacker_db_configs.into_iter() {
-                let new_config = match StackerDBConfig::from_smart_contract(
-                    chainstate,
-                    sortdb,
-                    &stackerdb_contract_id,
-                ) {
-                    Ok(config) => config,
-                    Err(e) => {
-                        warn!(
-                            "Failed to load StackerDB config for {}: {:?}",
-                            &stackerdb_contract_id, &e
-                        );
-                        StackerDBConfig::noop()
-                    }
-                };
+                let new_config =
+                    if stackerdb_contract_id == boot_code_id(MINERS_NAME, chainstate.mainnet) {
+                        // .miners contract -- directly generate the config
+                        let miners_config =
+                            match NakamotoChainState::make_miners_stackerdb_config(sortdb) {
+                                Ok(config) => config,
+                                Err(e) => {
+                                    warn!("Failed to generate .miners config: {:?}", &e);
+                                    continue;
+                                }
+                            };
+                        miners_config
+                    } else {
+                        // normal stackerdb contract
+                        match StackerDBConfig::from_smart_contract(
+                            chainstate,
+                            sortdb,
+                            &stackerdb_contract_id,
+                        ) {
+                            Ok(config) => config,
+                            Err(e) => {
+                                warn!(
+                                    "Failed to load StackerDB config for {}: {:?}",
+                                    &stackerdb_contract_id, &e
+                                );
+                                StackerDBConfig::noop()
+                            }
+                        }
+                    };
```

Comment:
```suggestion
                let new_config_res =
                    if stackerdb_contract_id == boot_code_id(MINERS_NAME, chainstate.mainnet) {
                        // .miners contract -- directly generate the config
                        NakamotoChainState::make_miners_stackerdb_config(sortdb)
                    } else {
                        // normal stackerdb contract
                        StackerDBConfig::from_smart_contract(
                            chainstate,
                            sortdb,
                            &stackerdb_contract_id,
                        )
                     };
                let new_config = new_config_res
                    .unwrap_or_else(|e| {
                                warn!(
                                    "Failed to load StackerDB config";
                                    "contract" => %stackerdb_contract_id,
                                    "err" => ?e,
                                );
                                StackerDBConfig::noop()
                     });
```

This suggestion changes the error behavior for the miners contract to use the noop db config. Is it necessary for `continue` to be used here? If so, why is it different in this regard than other stackerdb configs?

Reply:
I actually wonder if this should error if it fails to make a MINERS stackerdb config...If we continue, we will have a more nebulous error down the line due to the missing .miners stacker db no? Otherwise, I think using a noop is a better option as well to just continuing.

Reply:
Ah, this is an oversight. If we fail to generate the `.miners` StackerDB config, then we should use the previous one instead of simply refusing to synthesize it.  Then at least the chain can continue.

This error path is reachable only from database errors in the SortitionDB, however.  If it fails, it may be best to just shut down the node since it's unclear how those errors could ever be recoverable.

Reply:
I chose to have it error as I think this would be due to sortition DB errors which would indicate a much bigger problem